pkg/proxy/nftables/helpers_test.go

```diff
@@ -202,11 +202,12 @@ func (tracer *nftablesTracer) addressMatches(ipStr, not, ruleAddress string) boo
 	}
 }
 
-func (tracer *nftablesTracer) noneAddressesMatch(ipStr, ruleAddress string) bool {
+func (tracer *nftablesTracer) addressMatchesSet(ipStr, not, ruleAddress string) bool {
 	ruleAddress = strings.ReplaceAll(ruleAddress, " ", "")
 	addresses := strings.Split(ruleAddress, ",")
 	for _, address := range addresses {
-		if tracer.addressMatches(ipStr, "", address) {
+		match := tracer.addressMatches(ipStr, "", address)
+		if (match && not == "!= ") || (!match && not == "") {
```

Comment:
oof. maybe change the `not string` arg to `wantMatch bool` (which the caller would just pass `not == ""` for) and then you can say `if match == wantMatch {`